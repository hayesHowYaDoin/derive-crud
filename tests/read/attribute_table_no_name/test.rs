use derive_crud::Read;

#[derive(Read)]
#[crud_table]
struct Test {
    #[crud_id]
    id: i64,
    name: String,
}

#[tokio::main]
async fn main() {
    let _result = Test::read(1).await.unwrap(); // Test should fail before unwrap
}
