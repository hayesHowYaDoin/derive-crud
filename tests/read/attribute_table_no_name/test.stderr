error: A struct must be annotated with `#[crud_table("table_name")]` to specify the database table
 --> tests/read/attribute_table_no_name/test.rs:4:1
  |
4 | / #[crud_table]
5 | | struct Test {
6 | |     #[crud_id]
7 | |     id: i64,
8 | |     name: String,
9 | | }
  | |_^

error[E0599]: no function or associated item named `read` found for struct `Test` in the current scope
  --> tests/read/attribute_table_no_name/test.rs:16:19
   |
5  | struct Test {
   | ----------- function or associated item `read` not found for this struct
...
16 |     let _ = Test::read(&pool, 1);
   |                   ^^^^ function or associated item not found in `Test`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following traits define an item `read`, perhaps you need to implement one of them:
           candidate #1: `AsyncReadExt`
           candidate #2: `digest::XofReader`
           candidate #3: `futures_util::io::AsyncReadExt`
           candidate #4: `sqlx_core::net::socket::Socket`
           candidate #5: `std::io::Read`
